src/NuGet.Core/NuGet.Protocol/Plugins/Messages/GetCredentialsResponse.cs

```diff
@@ -28,6 +29,9 @@ public sealed class GetCredentialsResponse
         /// </summary>
         public string Username { get; }
 
+        public IReadOnlyList<string> AuthenticationTypes { get; }
```

Comment:
I get that this would not break the serialization but it feels very weird. 
We'd need to call it out explicitly in the spec. 

//cc @dtivel 
Can you please take a look at this?

Reply:
How is this weird? This is necessary to propagate the authentication information NuGet has, and it mirrors the communication going the other direction.